packages/react-server/src/ReactFizzServer.js

```diff
@@ -2542,15 +2542,22 @@ function trackPostpone(
 
     const children: Array<ReplayNode> = [];
     if (boundaryKeyPath === keyPath && task.childIndex === -1) {
-      // Since we postponed directly in the Suspense boundary we can't have written anything
-      // to its segment. Therefore this will end up becoming the root segment.
-      segment.id = boundary.rootSegmentID;
+      // Assign ID
+      if (segment.id === -1) {
+        if (segment.parentFlushed) {
+          // If this segment's parent was already flushed, it means we really just
+          // skipped the parent and this segment is now the root.
+          segment.id = boundary.rootSegmentID;
+        } else {
+          segment.id = request.nextSegmentId++;
+        }
+      }
```

Comment:
This code previously assumed we were inlining in the other place.

Assigning the ID here is not actually necessary because what will happen otherwise is that we'll generate an empty segment that doesn't get used with an unused id and then write into the root segment.

However, this is more conceptually correct and makes things work as it should - albeit with extra unnecessary segments.